File: CultureLinkCRM/CultureLinkCRM.Web/Modules/CrmDB/Person/PersonColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace CultureLinkCRM.CrmDB.Columns;

[ColumnsScript("CrmDB.Person")]
[BasedOnRow(typeof(PersonRow), CheckNames = true)]
public class PersonColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int PersonId { get; set; }
    [EditLink]
    public string FirstName { get; set; }
    public string LastName { get; set; }

    public string GenderName { get; set; }
}

File: CultureLinkCRM/CultureLinkCRM.Web/Modules/ServerTypes/CrmDB/PersonColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PersonRow } from "./PersonRow";

export interface PersonColumns {
    PersonId: Column<PersonRow>;
    FirstName: Column<PersonRow>;
    LastName: Column<PersonRow>;
    GenderName: Column<PersonRow>;
}

export class PersonColumns extends ColumnsBase<PersonRow> {
    static readonly columnsKey = 'CrmDB.Person';
    static readonly Fields = fieldsProxy<PersonColumns>();
}